### PR TITLE
Add thrown exception return to allow more tests

### PR DESCRIPTION
**Pull request by emri99**
Usage sample:

``` php
$test = function() {
    throw \UserException('user-id', 'Unknown user');
}
$userException = $this->assertException($test, '\UserException', null, 'Unknown user');
$this->assertEquals('user-id', $userException->getUserId());
```

### CODE REVIEW

**Comment by schnittstabil** [comment]
@emri99 Could you also add an example to the `readme.md`?

File: tests/AssertExceptionClassTest.php [hunk 1]
```diff
@@ -12,63 +12,63 @@ public function testDefault()
 		$test = function() {
 			throw new Exception();
 		};
-		AssertException::assertException($test, null);
+		$this->assertInstanceOf(Exception::class, AssertException::assertException($test, null));
```
Comment (schnittstabil) [inline tests/AssertExceptionClassTest.php]: IMO, the return type should not be tested in every test case:
If `assertException` does not return the expected type, then every test will fail – not very useful.

I'd suggest to test that in a single place, e.g. `testReturnType()`.

**Comment by emri99** [comment]
@schnittstabil Thanks for this advice
readme & tests updated ;)
let me know if you want others changes

File: tests/AssertExceptionClassTest.php [hunk 2]
```diff
@@ -71,4 +71,13 @@ public function testInterface()
 		AssertException::assertException($test, MyExceptionInterface::class);
 	}
 
+	public function testReturnType()
+	{
+	    $exception = new MyException();
```
Comment (schnittstabil) [inline tests/AssertExceptionClassTest.php]: CS: some spaces have crept into the indention

**Comment by schnittstabil** [comment]
> let me know if you want others changes

Don't get me wrong, I've reviewed your PR because I'd love to see it landed – @VladaHejda is the product owner…

File: tests/AssertExceptionClassTest.php (hunk 2 above)
Comment (emri99) [reply tests/AssertExceptionClassTest.php]: Thx, missed this one ;)

**Comment by schnittstabil** [comment]
@VladaHejda lgtm, I think it would be a great enhancement

**Comment by VladaHejda** [comment]
return added in version `1.1.0`

**Comment by schnittstabil** [comment]
@VladaHejda Awesome, thanks!

**Comment by emri99** [comment]
@VladaHejda Unfortunalely, this update is concurrent with requiring PHP7.
Would you consider creating a version 1.0.1 just merging this proposal with 1.0.0 ?
Thx :)

**Comment by schnittstabil** [comment]
@VladaHejda I believe the correct [Semantic Versioning](http://semver.org/) have to be:

* For PHP 7.1 compatibility require version `3.0.0`.
* For PHP 7 compatibility require version `2.0.0`.
* For PHP 5 compatibility require version `1.0.0`.

Thus, adding this proposal could be released as `1.1.0`.

**Comment by VladaHejda** [comment]
@emri99 caught exception is now returned in version `1.0.1` ;)

@schnittstabil hm, as proposed here http://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api semver does not order to increment major version whenever it's increased library dependency. However requiring higher PHP major version is quite bigger issue, thus it might deserve major version increase - mainly now there can't be new minor releases for older PHP versions.

Anyway, it has been once released as is, so it will remain.

**Pull request closed without merging**